gapis/api/vulkan/api/coherent_memory.api

```diff
@@ -57,25 +57,24 @@ sub void readCoherentMemory(ref!DeviceMemoryObject memory, VkDeviceSize readOffs
 }
 
 sub void readMemoryInBuffer(ref!BufferObject buffer, VkDeviceSize readOffset, VkDeviceSize readSize) {
-  memPieces := getBufferBoundMemoryPiecesInRange(buffer, readOffset, readSize)
-  for _, _, mp in memPieces {
-    readCoherentMemory(mp.DeviceMemory, mp.MemoryOffset, mp.Size)
-    read(mp.DeviceMemory.Data[mp.MemoryOffset:mp.MemoryOffset+mp.Size])
+  if (!buffer.VulkanHandle in ProcessedBuffers.val) {
+    memPieces := getBufferBoundMemoryPiecesInRange(buffer, readOffset, readSize)
+    for _, _, mp in memPieces {
+      readCoherentMemory(mp.DeviceMemory, mp.MemoryOffset, mp.Size)
+      read(mp.DeviceMemory.Data[mp.MemoryOffset:mp.MemoryOffset+mp.Size])
+    }
+    ProcessedBuffers.val[buffer.VulkanHandle] = true
   }
 }
 
 sub void readMemoryInBufferBindings(map!(u32, ref!VkDescriptorBufferInfo) bufferBindings, map!(u32, VkDeviceSize) bufferBindingOffsets) {
   for i in (0 .. len(bufferBindings)) {
     bufferBinding := bufferBindings[as!u32(i)]
     if bufferBinding.Buffer != as!VkBuffer(0) {
-      if (bufferBinding.Buffer in Buffers) {
-        offset := switch as!u32(i) in bufferBindingOffsets {
-          case true:
-            bufferBindingOffsets[as!u32(i)]
-          case false:
-            bufferBinding.Offset
+      if !(bufferBinding.Buffer in ProcessedBuffers.val) {
+        if (bufferBinding.Buffer in Buffers) {
+          readMemoryInBuffer(Buffers[bufferBinding.Buffer], 0, 0xFFFFFFFFFFFFFFFF)
```

Comment:
I have removed these changes. I think they will break the dependency graph in subtle ways, and I want to avoid that. Reads are only necessary in tracing if we are dealing with coherent memory. I will re-work part of this to see if we can do better without breaking dependency graph stuff.